app/src/main/java/com/quran/labs/androidquran/widgets/ForceCompleteTextView.java

```diff
@@ -46,23 +52,39 @@ protected void onFocusChanged(boolean focused, int direction, Rect previouslyFoc
     if (focused) {
       performFiltering(getText(), 0);
     } else {
-      // TODO create relevant listener name, such as onSelectChoice
-      AdapterView.OnItemClickListener listener = getOnItemClickListener();
-      if (listener != null)
-        listener.onItemClick(null, null, -1, -1);
+      onForceComplete(-1, AdapterView.INVALID_ROW_ID);
     }
   }
 
   /**
-   * Sets the listener that will be notified when the user clicks an item in the drop down list,
-   * user leaves without clicking, or when this view is attached to window. The two latter cases you
-   * use to force the completion, the listener will be called with position argument set to -1 and
-   * view argument set to null.
+   * Sets the listener that will be called to do force completion ({@link #onForceComplete(int,
+   * long)}).
+   */
+  public void setOnForceCompleteListener(@Nullable OnForceCompleteListener l) {
+    this.onForceCompleteListener = l;
+    post(() -> {
+      if (!isFocused()) {
+        onForceComplete(-2, AdapterView.INVALID_ROW_ID);
```

Comment:
There is no real reason, just to make them different. I don't know if it will be useful in the future.